src/types/message-types.ts

```diff
@@ -70,3 +70,22 @@ export type RangeFilter = (GT | LT) & Partial<GT> & Partial<LT>;
 export type Filter = {
   [property: string]: EqualFilter | OneOfFilter | RangeFilter
 };
+
+export type Pagination = {
+  offset?: number
```

Comment:
I am actually leaning towards cursor based pagination. Any reason why you lean towards offset based?

Reply:
We probably will be asked to support both if we are successful, which would be a great problem to have. In the meantime though, I prefer to prioritize cursor based pagination mainly because:

1. yes, more performant, but also
2. the records retrieved is not impacted by records being added or removed in real-time during retrieval of pages: with offset pagination, you can miss records if records are being deleted and see duplicate records if records are being added in real-time.

Adding @csuwildcat and @diehuxx to see if they have their own preferences.